ADR/00xx-component-dependencies.md

```diff
@@ -0,0 +1,179 @@
+
+### Integration-service and PR Groups
+
+* When [integration-service] notices a build from a PR (PR#2) that it detects is in the same group as a build from another PR (PR#1), it will perform testing but it will perform that process with some modifications. It will:
+  * Construct the Snapshot for the test using the image from the triggering build pipeline of PR #2 as well as the image found on the latest build pipeline associated with PR#1. This lets a user test PR #2 including unmerged content from PR #1. It will construct the Snapshot from the latest builds of all PRs in the PR Group.
+  * Post the test results back on PR#1 in addition to the results it would normally post to PR#2. It will post the test results back to all PRs in the PR Group.
+
+### Build-service and PR Groups
+
+* When [build-service] responds to the build of a PR (PR #1) and propagates the digest from one Component to another as a PR (PR#2). It follows the declared `depends on` references to know which other repos should receive an update PR. It marks the PR that is submits (PR #2) as being in the same PR Group as the triggering PR (PR #1). This enables pre-merge testing of both changes.
+* When [build-service] submits PR #2:
+  * It marks it as "Draft" and it includes a reference to the triggering PR (PR #1) in the description of the automatically submitted PR (PR #2), giving the user some indication that it should not be merged out of order.
+  * It checks to see if the two Components (source and destination as determined by the `depends on` references) are in the same git repository (a monorepo) and if PR #1 contains any changes to the `context` directory for the destination Component. If it does, then it create a new commit copying the code changes from that context directory in PR #1 to that context directory in PR #2, and it bases the commit that updates the digest reference on top of this synthetic commit.
+
+## Applied to Use Cases
+
+Let's apply the architecture to some use cases, and see how it plays out:
+
+### Common Parent Image
+
+Scenario: an application image depends on a common parent image. The user has 1 Application, and 2 Components. One of them is the parent image. The other image is built FROM that image.
+
+* The child image Component declares that it `depends on` the parent image Component, by way of the new field on the Component CR.
+* [integration-service] will always skip testing for parent image update builds, will never promote them, or use them to initiate Releases, but it will promote them to the global candidate list.
+* [build-service] will propagate digest references as a PR to the child image Component repo, by analyzing the `depends on` fields of all other Components in the Application. The PR that it files must be submitted in such a way that it appears in the same PR Group as the triggering PR submitted by a user.
```

Comment:
again 'The PR that it files must be submitted in such a way that it appears in the same PR Group'

you mean use the same slash prefixed name for git branch right?

I'd rephrase it and also use it everywhere